drivers/ethernet/dwc_xgmac/eth_dwc_xgmac.c

```diff
@@ -0,0 +1,1943 @@
+	uint32_t reg_val;
+	mem_addr_t ioaddr = get_reg_base_addr(dev);
+	mem_addr_t reg_addr;
+
+	for (dma_chnl = 0; dma_chnl < max_dma_chnl; dma_chnl++) {
+		tx_desc_meta = (struct xgmac_dma_tx_desc_meta *)&data->tx_desc_meta[dma_chnl];
+		rx_desc_meta = (struct xgmac_dma_rx_desc_meta *)&data->rx_desc_meta[dma_chnl];
+
+		reg_addr = (mem_addr_t)(ioaddr + XGMAC_DMA_CHNLx_BASE_ADDR_OFFSET(dma_chnl) +
+					DMA_CHx_CONTROL_OFST);
+		reg_val = DMA_CHx_CONTROL_SPH_SET(dma_chnl_cfg->sph) |
+			  DMA_CHx_CONTROL_PBLX8_SET(dma_chnl_cfg->pblx8) |
+			  DMA_CHx_CONTROL_MSS_SET(dma_chnl_cfg->mss);
+		sys_write32(reg_val, reg_addr);
+
+		reg_addr = (ioaddr + XGMAC_DMA_CHNLx_BASE_ADDR_OFFSET(dma_chnl) +
+			    DMA_CHx_TX_CONTROL_OFST);
+		reg_val = DMA_CHx_TX_CONTROL_TXPBL_SET(dma_chnl_cfg->txpbl) |
+			  DMA_CHx_TX_CONTROL_TSE_SET(dma_chnl_cfg->tse) |
+			  DMA_CHx_TX_CONTROL_RESERVED_OSP_SET(dma_chnl_cfg->osp);
+		sys_write32(reg_val, reg_addr);
+
+		reg_addr = (ioaddr + XGMAC_DMA_CHNLx_BASE_ADDR_OFFSET(dma_chnl) +
+			    DMA_CHx_RX_CONTROL_OFST);
+		reg_val = DMA_CHx_RX_CONTROL_RPF_SET(1u) |
+			  DMA_CHx_RX_CONTROL_RXPBL_SET(dma_chnl_cfg->rxpbl) |
+			  DMA_CHx_RX_CONTROL_RBSZ_SET(CONFIG_NET_BUF_DATA_SIZE);
+		sys_write32(reg_val, reg_addr);
+
+		reg_addr = (ioaddr + XGMAC_DMA_CHNLx_BASE_ADDR_OFFSET(dma_chnl) +
+			    DMA_CHx_TXDESC_LIST_HADDRESS_OFST);
+		reg_val = DMA_CHx_TXDESC_LIST_HADDRESS_TDESHA_SET(tx_desc_meta->desc_list_addr >>
+								  32u);
+		sys_write32(reg_val, reg_addr);
+
+		reg_addr = (ioaddr + XGMAC_DMA_CHNLx_BASE_ADDR_OFFSET(dma_chnl) +
+			    DMA_CHx_TXDESC_LIST_LADDRESS_OFST);
+		reg_val = tx_desc_meta->desc_list_addr;
+		sys_write32(reg_val, reg_addr);
+
+		reg_addr = (ioaddr + XGMAC_DMA_CHNLx_BASE_ADDR_OFFSET(dma_chnl) +
+			    DMA_CHx_RXDESC_LIST_HADDRESS_OFST);
+		reg_val = rx_desc_meta->desc_list_addr >> 32u;
+		sys_write32(reg_val, reg_addr);
+
+		reg_addr = (ioaddr + XGMAC_DMA_CHNLx_BASE_ADDR_OFFSET(dma_chnl) +
+			    DMA_CHx_RXDESC_LIST_LADDRESS_OFST);
+		reg_val = rx_desc_meta->desc_list_addr;
+		sys_write32(reg_val, reg_addr);
+
+		reg_addr = (ioaddr + XGMAC_DMA_CHNLx_BASE_ADDR_OFFSET(dma_chnl) +
+			    DMA_CHx_TXDESC_TAIL_LPOINTER_OFST);
+		reg_val = DMA_CHx_TXDESC_TAIL_LPOINTER_TDT_SET(tx_desc_meta->desc_tail_addr);
+		sys_write32(reg_val, reg_addr);
+
+		reg_addr = (ioaddr + XGMAC_DMA_CHNLx_BASE_ADDR_OFFSET(dma_chnl) +
+			    DMA_CHx_RXDESC_TAIL_LPOINTER_OFST);
+		reg_val = DMA_CHx_RXDESC_TAIL_LPOINTER_RDT_SET(rx_desc_meta->desc_tail_addr);
+		sys_write32(reg_val, reg_addr);
+
+		reg_addr = (ioaddr + XGMAC_DMA_CHNLx_BASE_ADDR_OFFSET(dma_chnl) +
+			    DMA_CHx_TX_CONTROL2_OFST);
+		reg_val = DMA_CHx_TX_CONTROL2_TDRL_SET((dma_chnl_cfg->tdrl - 1u));
+		sys_write32(reg_val, reg_addr);
+
+		reg_addr = (ioaddr + XGMAC_DMA_CHNLx_BASE_ADDR_OFFSET(dma_chnl) +
+			    DMA_CHx_RX_CONTROL2_OFST);
+		reg_val = DMA_CHx_RX_CONTROL2_RDRL_SET((dma_chnl_cfg->rdrl - 1u));
+		sys_write32(reg_val, reg_addr);
+
```

Comment:
a one-liner comment per sys_write32() calls to tell which register is being set and what's being configured, would be more readable

Reply:
"I've added a comment before the register is being written."